jsonrpc/dispatcher.go

```diff
@@ -478,24 +478,24 @@ func (d *Dispatcher) decodeTxn(arg *txnArgs) (*types.Transaction, error) {
 	// set default values
 	if arg.From == nil {
 		arg.From = &types.ZeroAddress
-	}
-	if arg.Data != nil && arg.Input != nil {
-		return nil, fmt.Errorf("both input and data cannot be set")
-	}
-	if arg.Nonce == nil {
+		arg.Nonce = argUintPtr(0)
+	} else if arg.Nonce == nil {
 		// get nonce from the pool
 		nonce, err := d.getNextNonce(*arg.From, LatestBlockNumber)
 		if err != nil {
 			return nil, err
 		}
 		arg.Nonce = argUintPtr(nonce)
 	}
+
+	if arg.Data != nil && arg.Input != nil {
+		return nil, fmt.Errorf("both input and data cannot be set")
+	}
```

Comment:
This was not added, git diff can be weird sometimes.

Reply:
Looking at the `develop`, it seems it was moved after the nonce check.
On `develop` branch this check is occurring before.

Reply:
What do you think of moving this to be the first condition here?
That way we avoid fetching the nonce if these 2 are nil.

Reply:
👍  https://github.com/0xPolygon/polygon-sdk/pull/124/commits/0d86f9045d7b012a9d0aeed2d98b81dd1eadcc30